polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -268,6 +284,32 @@ $loading-panel-height: 53px;
     .TableHeadingSortIcon {
       opacity: 1;
     }
+
+    .TableHeadingSortIcon-heading-align-end {
+      width: var(--p-space-4);
+    }
+  }
+}
+
+.TableHeadingSortButton-heading-align-end {
+  position: relative;
+  left: var(--p-space-4);
+  margin-right: var(--p-space-4);
+
+  transition: all var(--p-duration-200) var(--p-ease-in) var(--p-duration-100);
+
+  &:hover,
+  &:focus {
+    left: 0;
+    margin-right: 0;
+  }
+}
+
+.TableHeadingSortButton-heading-align-end-previously-sorted {
+  animation: hide-sort-button var(--p-duration-200) var(--p-ease-in);
+
+  .TableHeadingSortIcon {
+    animation: hide-sort-button-icon var(--p-duration-200) var(--p-ease-in);
```

Comment:
These styles animate the width of the sort icon after an active, right-aligned, sortable column becomes inactive due to a click on another sortable column.